sdk/communication/azure-communication-common/src/test/java/com/azure/android/communication/common/CommunicationIdentifierSerializerTests.java

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.android.communication.common;
+
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+public class CommunicationIdentifierSerializerTests {
+
+    final String someId = "some id";
+    final String teamsUserId = "Teams user id";
+    final String fullId = "some lengthy id string";
+
+    @Test
```

Comment:
We also have to test all those Objects.RequireNonNull on deserialization, i.e. that we throw when properties are missing.

See here for the list of tests: https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/communication/Azure.Communication.Common/tests/CommunicationIdentifierSerializerTest.cs

Reply:
Tested in another test file, IdentifierSerialzationExceptionTests